spec/lib/openstax/auth/strategy_2_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'spec_helper'
+require 'openstax/auth/strategy_2'
+
+RSpec.describe OpenStax::Auth::Strategy2 do
+
+  SIGNATURE_PUBLIC_KEY = <<~SIGNATURE
+      -----BEGIN PUBLIC KEY-----
+      MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAxItufComL1S+j+U2JVr4
+      aeIPyZYJR8S3RcxXjlHgybrAsBf/zyAjJlOBIhVfrm9jUF1HKNMyiTKQehG5XBBW
+      /F8DQT5vGdBd4g9WIAmNU0E+symeF4X+mFIZ6dYwTMKtZxv1U0nkJ8xL2q4jCFVB
+      UMlQRmz9EffSz+fmXr9xGQj80HKahzciM6m2aspX096qUP90155qmLEayE2uhs5C
+      oAUbahA+VXS6ggsCUeVyog5Z1mC086d8r78ylr1y8IQ3aazdJE/mKxfqvu9S423h
+      wNzBP6Fp0n68ZjUdUirqAZEbSrioJgFLEzX8ef7XilTL9dKLSS1w09ErctAF+Tor
+      hwIDAQAB
+      -----END PUBLIC KEY-----
+  SIGNATURE
+
+  SECRETS = {
+    # these values copied from the Accounts secrets
+    strategy_2_cookie_name: "oxa_dev",
+    strategy_2_signature_public_key: SIGNATURE_PUBLIC_KEY,
+    strategy_2_encryption_private_key: 'RvGHVZ/kvzUAA5Z3t68+FNhuMCJxkzv+',
+    strategy_2_encryption_algorithm: 'dir',
+    strategy_2_encryption_method: 'A256GCM',
+    strategy_2_signature_algorithm: 'RS256'
+  }
+
+  before(:each) do
+    SECRETS.each do |name, value|
+      allow(OpenStax::Auth.configuration).to receive(name).and_return(value)
+    end
+  end
+
+  let(:mock_request) do
+    OpenStruct.new(
+      cookies: {
+        'oxa_dev' =>
+          "eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..tD56OCg6Un6gzkTy.eQAZrea_ORzHiFvg3rmEjxjmRXlqhXIfOYA55RQP9T" \
+          "uum2S7d-KqFQiNEVHvdOAz8Jt1UULezUllc1bJ42j7IiBlakUm8VlOIhS6n0XtobmuQkIV7nd_0_DNTn2uvmLC70Z9pGtgUOFm3q" \
+          "XtQa5DczQTuoTQ56-_M9uewyIYtj_B0H0bfjDNvsj92hf54K8o486B97qnvfGIkb7jXRUk3q6Aa_NtNIqbmCR9Tac29H9rn7CAcV" \
+          "TG2vzG-kfUwpxrA9A-LlkX4SX1LKzsD3TVQe-05Xv9cQdf8zArdeE_9KJGAdbRlM-DVK3ul2YBTy4z92uxY4cA7vtNsANb1ByNn7" \
+          "K5zEa4Mnb1OcxhhrPKTTkyVWtt4-w8GhmZl48kBoeQTWEEXtRlksabKe5RhHu3-i3dXvbWBp6ALXjEkAoKC-BDDjCUt_IOErp_g0" \
+          "G1CnD3aRU--lqvm2IJnKq1sncTd8qtFTm91MRPzg94O0-OHk7NohktEz3DtJjKeH0EdW98d_mon8OAf4xJDtXrADE-VxAMPhNzoF" \
+          "s6o2k4t3BJpIvUj9AGuAx46vkk7B0TeIAXFy8dhq6n5vvFdYnoih1BM47DnOv5DZtABlvQv5xJTfyN23jb-QDKG-AZ--zjtamtkT" \
+          "r_7GASXqbQy2xEw2QA0yQCUS6JhnRRCcrC913CU8uPtjMbzWoxkCZjCyxQkX1fcVddU9e3pmay9LZ4zZolVCOwWUp1TuEgYwSweN" \
+          "pR4WiwGiWelMhHSZ3QYKjJpGyIkzCSkn7ZQRLLTe0joU43figYs790TPx4waUfwi5r3AED6OSkfxTBsjgOR9DY6083CpCZ4N7lea" \
+          "XhsfepgwjiwzVw5TB4YGRg275AE4lZhdKf1lgS7OSk1S7NeMkv88ZDHnVIVAd0wiR9PZf36Ni48CArfC4btn6DT7cQURQOnQTQyi" \
+          "K-WvFfkEMdWX7_Z-GRG9CCnVIT3CBBZnvoIcCaUbVmXRqv0cFJmvfsmGsA.FJxz84tw7BCYCrwYeqLpdQ" \
+      }
+    )
+  end
+
+  describe "#decrypt" do
+    subject(:decoding) {
+      described_class.decrypt(mock_request)
+    }
+
+    it 'decrypts' do
+      expect(decoding['sub']['uuid']).to eq '1b2dc73a-a792-462b-9b0f-59bd22bac26d'
+      expect(decoding['sub']['is_test']).to be_falsey
```

Comment:
what is `sub`

Reply:
it was what was in the decoded cookie

Reply:
It's a standard JWT field but now I'm looking at the spec and I think I should have put just the uuid in there https://tools.ietf.org/html/rfc7519#section-4.1.2

Ugh... should I make a PR to rename "sub" to "user" and just put the uuid in "sub" from now on?

Reply:
I guess in the end it doesn't matter too much that sub doesn't follow the spec unless that causes some JWT library to explode, which I think is unlikely...